htdocs/router.php

```diff
@@ -70,6 +70,12 @@
     $_REQUEST['subtest']   = $getParams[3];
 
     include_once __DIR__ . "/main.php";
+} elseif (preg_match(
+    '#^api/.*#',
+    $url
+)) {
+    $endpoint = explode('?', $url)[0];
+    include_once __DIR__ . '/' . $endpoint;
```

Comment:
Missed the actual comment that I was here for in my review..

I couldn't get this to work. I'm still getting a 404 with the error "[Thu Sep 20 11:02:31 2018] ::1:65216 [404]: /api/v0.0.2/projects - Undefined error: 0" while trying to access `/api/v0.0.2/projects/` through the php built in web server

Reply:
By changing the first if statement to:

```
f ($request != '/'
    && file_exists(__DIR__ . $request)
    && $request != "/acknowledgements/"
    && preg_match('/^api\//', $url) !== 1
) { 
```

I was at least able to reach this part of the router, but I'm still not sure how this is supposed to work since there isn't a 1-1 mapping between file system layout and URL endpoints (in particular, the capitalization and file extensions differ)

Reply:
(and there's the parts that are dynamic in between like the IDs)

Reply:
@xlecours Could you comment here? I believe you did this part of the code.

Reply:
@driusan This PR is to test the current (v0.0.3-dev) api endpoints without any kind of url rewriting. `/api/v0.0.2/projects/` wont work with that. But `/api/v0.0.2/Projects.php?PrintProjects=true` will

Reply:
But there are no API endpoints without any kind of URL rewriting. The API is defined in terms of the URLs.